service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java

```diff
@@ -1,209 +1,288 @@
 package de.adorsys.psd2.sandbox.portal.testdata;
 
 import de.adorsys.psd2.sandbox.portal.testdata.domain.Account;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Amount;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Balance;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.PsuData;
 import de.adorsys.psd2.sandbox.portal.testdata.domain.Transaction;
 import java.math.BigDecimal;
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.Currency;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TestDataService {
 
-  public static final String PSU_ID = "PSU-1";
-  public static final String PSU_PASSWORD = "12345";
-  public static final String IBAN_GIRO = "DE94500105178833114935";
-  public static final String IBAN_SAVINGS = "DE96500105179669622432";
-  public static final Currency EUR = Currency.getInstance("EUR");
-  public static final String ACCOUNT_OWNER = "Melanie Klein";
-  public static final String ACCOUNT_ID_GIRO = "9b86539d-589b-4082-90c2-d725c019777f";
-  public static final String ACCOUNT_ID_SAVINGS = "d460057b-053a-490a-a36e-c0c8afb735e9";
+  private static final Currency EUR = Currency.getInstance("EUR");
 
-  private HashMap<String, Account> accounts;
-  private HashMap<String, HashMap<String, Transaction>> transactions;
+  private HashMap<String, PsuData> psuMap;
 
+  /**
+   * Default Constructor.
+   */
   public TestDataService() {
-    this.accounts = createAccounts();
-    this.transactions = createTransactions();
+    HashMap<String, PsuData> map = new HashMap<>();
+
+    PsuData psuSuccessful = initPsuSuccessfull();
+
+    map.put(psuSuccessful.getPsuId(), psuSuccessful);
+
+    this.psuMap = map;
   }
 
-  public List<Transaction> getTransactions(String accountId) {
-    return new ArrayList<>(transactions.get(accountId).values());
+  public PsuData getPsu(String psuId) {
+    return psuMap.get(psuId);
   }
 
-  public Transaction getTransactionDetails(String accountId, String transactionId) {
-    return transactions.get(accountId).get(transactionId);
+  /**
+   * Returns Psu-Id by Iban.
+   *
+   * @param iban Iban
+   * @return String Psu-Id
+   */
+  public String getPsuByIban(String iban) {
+    for (Map.Entry<String, PsuData> currentPsu : psuMap.entrySet()) {
+      String currentPsuId = currentPsu.getValue().getPsuId();
+
+      List<Account> result = currentPsu.getValue().getAccounts().values().stream()
+          .filter(account -> account.getIban().equals(iban))
+          .collect(Collectors.toList());
+
+      if (result.size() > 0) {
+        return currentPsuId;
+      }
+    }
+    return "";
```

Comment:
Throw exceptions instead of returning defaults or return `Optional`.

Reply:
I return an Optional now